builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -336,15 +340,6 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 							Type:     schema.TypeString,
 							Optional: true,
 							ForceNew: true,
-							Default:  "eager_zeroed",
```

Comment:
FYI, I removed the default here because the default + ForceNew: true creates an interesting situation.  If the user were to not set the type, the .tfstate file would still end up with an entry for type = "eager_zeroed".  If you run terraform plan immediately after, it would tell you that you are attempting to alter the "type" state from "eager_zeroed" to "" (since your .tf script does not set "type").  With the addition of the ForceNew, this would blow away the VM.  Bad news bears.  It seems that Default values don't play nicely with Update scenarios.  I moved the Default setting into the Create() block, which is functionally equivalent.